### PR TITLE
Add pre-commit configuration and autoformat

### LINKED ISSUE #66: Configure pre-commit config
Suggested tool `black`, `isort` in #29, could also use `ruff`.

### PR DESCRIPTION
**Pull request by abhidg**
This PR adds pre-commit configuration. I have opted to use [`ruff`](https://docs.astral.sh/ruff/) as the linter, open to alternatives. Reason for choosing `ruff` over `flake8`, `black` is that it is one tool that combines most Python autoformatting and linting functions in one and is very fast.

- Add pre-commit configuration, fixes #66
- Initial pre-commit run with ruff autoformatting
- Add git blame ignore revisions file

### CODE REVIEW

File: .vscode/settings.json
Comment (tlestang) [inline .vscode/settings.json:1]: Do we need vs code specific config in the git repo?

**Review by tlestang — APPROVED** [review]
Ah you beat me to it!! I also went with ruff, so that's a yes for me :) thanks @abhidg

File: .vscode/settings.json
Comment (abhidg) [reply .vscode/settings.json:1]: Don't think so - removing!

**Comment by ljcolling** [comment]
Good choice with Ruff, and it kinda fits with the ethos of the project

**Pull request merged**